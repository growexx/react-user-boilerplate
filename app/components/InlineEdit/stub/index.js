export const TEST_IDS = {
  INPUT_VALUE: 'INPUT_VALUE',
  INPUT_EDIT: 'INPUT_EDIT',
  SAVE_BUTTON: 'save',
  CANCEL_BUTTON: 'cancel',
};
