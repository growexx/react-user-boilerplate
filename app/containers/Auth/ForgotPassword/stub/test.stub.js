export const TEST_IDS = {
  RESET_PASSWORD: 'RESET_PASSWORD',
};
