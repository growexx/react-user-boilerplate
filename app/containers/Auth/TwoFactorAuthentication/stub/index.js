export const stubUserData = elseCase => {
  if (elseCase) {
    return {
      docs: [],
    };
  }
  return {
    docs: [
      {
        id: '0',
      },
    ],
  };
};
