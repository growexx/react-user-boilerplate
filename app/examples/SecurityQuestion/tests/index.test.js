import React from 'react';
import { render, wait, fireEvent } from 'react-testing-library';
import request from 'utils/request';
import 'jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { ConnectedRouter } from 'connected-react-router/immutable';
import { createMemoryHistory } from 'history';
// import ReactDOM from 'react-router-dom';

import RegisterQuestion from '../RegisterQuestion/index';
import ResetPassword from '../ResetPassword/index';
import { registerSecurityQuestion } from '../stub';
import configureStore from '../../../configureStore';

jest.mock('utils/request');

export async function selectOption(
  container,
  optionText,
  getByText,
  findByText,
  keyDownEvent,
) {
  const placeholder = getByText(container, 'Select...');
  fireEvent.keyDown(placeholder, keyDownEvent);
  await findByText(container, optionText);
  fireEvent.click(getByText(container, optionText));
}

describe('<ResetPassword />', () => {
  it('should match snapshot', () => {
    const history = createMemoryHistory();
    const store = configureStore({}, history);

    // it('should render a div', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <ConnectedRouter history={history}>
            <ResetPassword />
          </ConnectedRouter>
        </IntlProvider>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
it('should match snapshot', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  // it('should render a div', () => {
  const { container } = render(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ConnectedRouter history={history}>
          <RegisterQuestion />
        </ConnectedRouter>
      </IntlProvider>
    </Provider>,
  );
  expect(container.firstChild).toMatchSnapshot();
});
// });
// it('shows a render list', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<RegisterQuestion />, div);
//   expect(div.innerHTML).toContain('Set Security Question');

//   ReactDOM.unmountComponentAtNode(div);
// });

it('should render an <StyledSecurityQuestion> tag', () => {
  const {
    container: { firstChild },
  } = render(<RegisterQuestion />);
  expect(firstChild.tagName).toEqual('DIV');
});

it('should render an <StyledSecurityQuestion> tag', () => {
  const {
    container: { firstChild },
  } = render(<RegisterQuestion />);
  expect(firstChild.tagName).toMatchSnapshot();
});
// it('should render an <StyledSecurityQuestion> tag', () => {
//   const {
//     container: { firstChild },
//   } = render(<ResetPassword />);
//   expect(firstChild.tagName).toMatchSnapshot();
// });

it('should adopt any attribute', () => {
  const { container } = render(<RegisterQuestion attribute="test" />);
  expect(container.firstChild.hasAttribute('attribute')).toBe(false);
});

it('should render and match the snapshot', () => {
  request.mockImplementation(() => Promise.resolve({ status: 1 }));
  const {
    container: { firstChild },
  } = render(<RegisterQuestion />);
  expect(firstChild).toMatchSnapshot();
});

it('should render and match the snapshot', async () => {
  request.mockImplementation(registerSecurityQuestion);
  const {
    container: { firstChild },
  } = render(<RegisterQuestion />);

  await wait(() => expect(request).toHaveBeenCalledTimes(0));
  expect(firstChild).toMatchSnapshot();
});
// });

it('should call submit function call', async () => {
  request.mockImplementationOnce(() => {
    registerSecurityQuestion({ question1: 'abc', answer: 'abc' });
  });
  const { getAllByRole, getByTestId } = render(<RegisterQuestion />);
  await wait(() => expect(getAllByRole('combobox')[0]).toBeTruthy());
  const ques = getAllByRole('combobox');
  fireEvent.mouseDown(ques[0]);
  fireEvent.click(
    document.querySelectorAll('.ant-select-item-option-content')[0],
  );

  const eventObject = {
    target: {
      value: 'test',
      name: 'test',
    },
  };
  fireEvent.change(getByTestId('answer1'), eventObject);
  fireEvent.change(getByTestId('answer2'), eventObject);
  fireEvent.change(getByTestId('answer3'), eventObject);
  const submitbtn = getByTestId('submit_question');
  submitbtn.setAttribute('disabled', false);

  expect(submitbtn).toBeDisabled();
  await wait(() => fireEvent.click(submitbtn));
  expect(request).toHaveBeenCalledTimes(0);
});
