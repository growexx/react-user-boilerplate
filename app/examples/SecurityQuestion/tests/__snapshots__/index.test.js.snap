// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetPassword /> should match snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .title-main {
  font-weight: 700;
  font-size: 32px;
  color: #4d186e;
}

.c1 .question-form {
  max-width: 460px;
}

.c0 {
  font-family: 'Montserrat',sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  height: 100vh;
}

.c0 .header {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
}

.c0 .link-reset {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .link-security {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: inline-block;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .registrationSideContainer {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
}

.c0 .registrationSideContainer .title {
  text-align: center;
  font-size: 42px;
  font-weight: 700;
  margin: 0;
}

.c0 .registrationSideContainer .subTitle {
  width: 60%;
  text-align: center;
  margin: 0;
}

.c0 .registrationSideContainer button {
  margin-top: 35px;
  height: 40px;
  width: 160px;
  background: transparent;
  border: 2px solid #fff;
  border-radius: 50px;
  text-align: center;
  color: #fff;
}

.c0 .registrationSideContainer button:hover {
  color: #fff;
}

.c0 .main {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0.7;
  -ms-flex: 0.7;
  flex: 0.7;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .resetPassword {
  font-weight: 700;
  font-size: 42px;
  color: #4d186e;
  margin: unset;
}

.c0 .sideContainer {
  background: #4d186e;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0.3;
  -ms-flex: 0.3;
  flex: 0.3;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .security-main {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0.7;
  -ms-flex: 0.7;
  flex: 0.7;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .security-main .title {
  font-weight: 700;
  font-size: 32px;
  color: #4d186e;
}

.c0 .title-reset {
  font-weight: 700;
  font-size: 32px;
  color: #4d186e;
}

<div
  class="c0"
>
  <div
    class="sideContainer"
  >
    <div
      class="loginContainer"
    >
      <div
        class="registrationSideContainer"
      >
        <p
          class="title"
        >
          <span>
            New Here!
          </span>
        </p>
        <p
          class="subTitle"
        >
          <span>
            To join us, please sign up with your personal info
          </span>
        </p>
        <a
          href="/register"
        >
          <button
            class="ant-btn"
            type="button"
          >
            <span>
              SIGN UP
            </span>
          </button>
        </a>
      </div>
    </div>
  </div>
  <div
    class="security-main"
  >
    <p
      class="title-reset"
    >
      Please Verify your Security question
    </p>
    <div
      class="c1"
    >
      <button
        class="ant-btn ant-btn-primary"
        data-testid="submit_question"
        disabled=""
        type="button"
      >
        <span>
          Submit
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`should match snapshot 1`] = `
.c1 {
  width: 100%;
}

.c1 .input-select {
  width: 100%;
}

.c1 .answer {
  margin: 1rem 0px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .title-main {
  font-weight: 700;
  font-size: 32px;
  color: #4d186e;
}

.c0 .question-form {
  max-width: 460px;
}

<div
  class="c0"
>
  <div
    class="container"
  >
    <p
      class="title-main"
    >
      Set Security Question
    </p>
    <div
      class="question-form"
    >
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="ant-select input-select ant-select-single ant-select-show-arrow"
            data-testid="question1"
            name="question1"
          >
            <div
              class="ant-select-selector"
            >
              <span
                class="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-haspopup="listbox"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autocomplete="off"
                  class="ant-select-selection-search-input"
                  id="rc_select_TEST_OR_SSR"
                  readonly=""
                  role="combobox"
                  style="opacity: 0;"
                  type="search"
                  unselectable="on"
                  value=""
                />
              </span>
              <span
                class="ant-select-selection-item"
                title="-- Choose a Security Question --"
              >
                -- Choose a Security Question --
              </span>
            </div>
            <span
              aria-hidden="true"
              class="ant-select-arrow"
              style="user-select: none;"
              unselectable="on"
            >
              <span
                aria-label="down"
                class="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <input
            class="ant-input answer"
            data-testid="answer1"
            name="answer1"
            placeholder="Answer"
            type="text"
            value=""
          />
        </div>
        <div
          class="c1"
        >
          <div
            class="ant-select input-select ant-select-single ant-select-show-arrow"
            data-testid="question2"
            name="question2"
          >
            <div
              class="ant-select-selector"
            >
              <span
                class="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-haspopup="listbox"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autocomplete="off"
                  class="ant-select-selection-search-input"
                  id="rc_select_TEST_OR_SSR"
                  readonly=""
                  role="combobox"
                  style="opacity: 0;"
                  type="search"
                  unselectable="on"
                  value=""
                />
              </span>
              <span
                class="ant-select-selection-item"
                title="-- Choose a Security Question --"
              >
                -- Choose a Security Question --
              </span>
            </div>
            <span
              aria-hidden="true"
              class="ant-select-arrow"
              style="user-select: none;"
              unselectable="on"
            >
              <span
                aria-label="down"
                class="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <input
            class="ant-input answer"
            data-testid="answer2"
            name="answer2"
            placeholder="Answer"
            type="text"
            value=""
          />
        </div>
        <div
          class="c1"
        >
          <div
            class="ant-select input-select ant-select-single ant-select-show-arrow"
            data-testid="question3"
            name="question3"
          >
            <div
              class="ant-select-selector"
            >
              <span
                class="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-haspopup="listbox"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autocomplete="off"
                  class="ant-select-selection-search-input"
                  id="rc_select_TEST_OR_SSR"
                  readonly=""
                  role="combobox"
                  style="opacity: 0;"
                  type="search"
                  unselectable="on"
                  value=""
                />
              </span>
              <span
                class="ant-select-selection-item"
                title="-- Choose a Security Question --"
              >
                -- Choose a Security Question --
              </span>
            </div>
            <span
              aria-hidden="true"
              class="ant-select-arrow"
              style="user-select: none;"
              unselectable="on"
            >
              <span
                aria-label="down"
                class="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <input
            class="ant-input answer"
            data-testid="answer3"
            name="answer3"
            placeholder="Answer"
            type="text"
            value=""
          />
        </div>
        <button
          class="ant-btn ant-btn-primary"
          data-testid="submit_question"
          disabled=""
          type="button"
        >
          <span>
            Submit
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render an <StyledSecurityQuestion> tag 1`] = `"DIV"`;

exports[`should render and match the snapshot 1`] = `
.c1 {
  width: 100%;
}

.c1 .input-select {
  width: 100%;
}

.c1 .answer {
  margin: 1rem 0px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .title-main {
  font-weight: 700;
  font-size: 32px;
  color: #4d186e;
}

.c0 .question-form {
  max-width: 460px;
}

<div
  class="c0"
>
  <div
    class="container"
  >
    <p
      class="title-main"
    >
      Set Security Question
    </p>
    <div
      class="question-form"
    >
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="ant-select input-select ant-select-single ant-select-show-arrow"
            data-testid="question1"
            name="question1"
          >
            <div
              class="ant-select-selector"
            >
              <span
                class="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-haspopup="listbox"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autocomplete="off"
                  class="ant-select-selection-search-input"
                  id="rc_select_TEST_OR_SSR"
                  readonly=""
                  role="combobox"
                  style="opacity: 0;"
                  type="search"
                  unselectable="on"
                  value=""
                />
              </span>
              <span
                class="ant-select-selection-item"
                title="-- Choose a Security Question --"
              >
                -- Choose a Security Question --
              </span>
            </div>
            <span
              aria-hidden="true"
              class="ant-select-arrow"
              style="user-select: none;"
              unselectable="on"
            >
              <span
                aria-label="down"
                class="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <input
            class="ant-input answer"
            data-testid="answer1"
            name="answer1"
            placeholder="Answer"
            type="text"
            value=""
          />
        </div>
        <div
          class="c1"
        >
          <div
            class="ant-select input-select ant-select-single ant-select-show-arrow"
            data-testid="question2"
            name="question2"
          >
            <div
              class="ant-select-selector"
            >
              <span
                class="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-haspopup="listbox"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autocomplete="off"
                  class="ant-select-selection-search-input"
                  id="rc_select_TEST_OR_SSR"
                  readonly=""
                  role="combobox"
                  style="opacity: 0;"
                  type="search"
                  unselectable="on"
                  value=""
                />
              </span>
              <span
                class="ant-select-selection-item"
                title="-- Choose a Security Question --"
              >
                -- Choose a Security Question --
              </span>
            </div>
            <span
              aria-hidden="true"
              class="ant-select-arrow"
              style="user-select: none;"
              unselectable="on"
            >
              <span
                aria-label="down"
                class="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <input
            class="ant-input answer"
            data-testid="answer2"
            name="answer2"
            placeholder="Answer"
            type="text"
            value=""
          />
        </div>
        <div
          class="c1"
        >
          <div
            class="ant-select input-select ant-select-single ant-select-show-arrow"
            data-testid="question3"
            name="question3"
          >
            <div
              class="ant-select-selector"
            >
              <span
                class="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-haspopup="listbox"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autocomplete="off"
                  class="ant-select-selection-search-input"
                  id="rc_select_TEST_OR_SSR"
                  readonly=""
                  role="combobox"
                  style="opacity: 0;"
                  type="search"
                  unselectable="on"
                  value=""
                />
              </span>
              <span
                class="ant-select-selection-item"
                title="-- Choose a Security Question --"
              >
                -- Choose a Security Question --
              </span>
            </div>
            <span
              aria-hidden="true"
              class="ant-select-arrow"
              style="user-select: none;"
              unselectable="on"
            >
              <span
                aria-label="down"
                class="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <input
            class="ant-input answer"
            data-testid="answer3"
            name="answer3"
            placeholder="Answer"
            type="text"
            value=""
          />
        </div>
        <button
          class="ant-btn ant-btn-primary"
          data-testid="submit_question"
          disabled=""
          type="button"
        >
          <span>
            Submit
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render and match the snapshot 2`] = `
.c1 {
  width: 100%;
}

.c1 .input-select {
  width: 100%;
}

.c1 .answer {
  margin: 1rem 0px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .title-main {
  font-weight: 700;
  font-size: 32px;
  color: #4d186e;
}

.c0 .question-form {
  max-width: 460px;
}

<div
  class="c0"
>
  <div
    class="container"
  >
    <p
      class="title-main"
    >
      Set Security Question
    </p>
    <div
      class="question-form"
    >
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="ant-select input-select ant-select-single ant-select-show-arrow"
            data-testid="question1"
            name="question1"
          >
            <div
              class="ant-select-selector"
            >
              <span
                class="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-haspopup="listbox"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autocomplete="off"
                  class="ant-select-selection-search-input"
                  id="rc_select_TEST_OR_SSR"
                  readonly=""
                  role="combobox"
                  style="opacity: 0;"
                  type="search"
                  unselectable="on"
                  value=""
                />
              </span>
              <span
                class="ant-select-selection-item"
                title="-- Choose a Security Question --"
              >
                -- Choose a Security Question --
              </span>
            </div>
            <span
              aria-hidden="true"
              class="ant-select-arrow"
              style="user-select: none;"
              unselectable="on"
            >
              <span
                aria-label="down"
                class="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <input
            class="ant-input answer"
            data-testid="answer1"
            name="answer1"
            placeholder="Answer"
            type="text"
            value=""
          />
        </div>
        <div
          class="c1"
        >
          <div
            class="ant-select input-select ant-select-single ant-select-show-arrow"
            data-testid="question2"
            name="question2"
          >
            <div
              class="ant-select-selector"
            >
              <span
                class="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-haspopup="listbox"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autocomplete="off"
                  class="ant-select-selection-search-input"
                  id="rc_select_TEST_OR_SSR"
                  readonly=""
                  role="combobox"
                  style="opacity: 0;"
                  type="search"
                  unselectable="on"
                  value=""
                />
              </span>
              <span
                class="ant-select-selection-item"
                title="-- Choose a Security Question --"
              >
                -- Choose a Security Question --
              </span>
            </div>
            <span
              aria-hidden="true"
              class="ant-select-arrow"
              style="user-select: none;"
              unselectable="on"
            >
              <span
                aria-label="down"
                class="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <input
            class="ant-input answer"
            data-testid="answer2"
            name="answer2"
            placeholder="Answer"
            type="text"
            value=""
          />
        </div>
        <div
          class="c1"
        >
          <div
            class="ant-select input-select ant-select-single ant-select-show-arrow"
            data-testid="question3"
            name="question3"
          >
            <div
              class="ant-select-selector"
            >
              <span
                class="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-haspopup="listbox"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autocomplete="off"
                  class="ant-select-selection-search-input"
                  id="rc_select_TEST_OR_SSR"
                  readonly=""
                  role="combobox"
                  style="opacity: 0;"
                  type="search"
                  unselectable="on"
                  value=""
                />
              </span>
              <span
                class="ant-select-selection-item"
                title="-- Choose a Security Question --"
              >
                -- Choose a Security Question --
              </span>
            </div>
            <span
              aria-hidden="true"
              class="ant-select-arrow"
              style="user-select: none;"
              unselectable="on"
            >
              <span
                aria-label="down"
                class="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <input
            class="ant-input answer"
            data-testid="answer3"
            name="answer3"
            placeholder="Answer"
            type="text"
            value=""
          />
        </div>
        <button
          class="ant-btn ant-btn-primary"
          data-testid="submit_question"
          disabled=""
          type="button"
        >
          <span>
            Submit
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
